dsl_parser/elements/imports.py

```diff
@@ -259,27 +287,33 @@ def build_ordered_imports_recursive(_current_parsed_dsl_holder,
                                                           resources_base_path,
                                                           _current_import)
             validate_namespace(namespace)
+            current_namespace = namespace
             if context_namespace:
                 if namespace:
-                    namespace = utils.generate_namespaced_value(
+                    current_namespace = utils.generate_namespaced_value(
                         context_namespace,
                         namespace)
                 else:
                     # In case a namespace was added earlier in the import
                     # chain.
-                    namespace = context_namespace
+                    current_namespace = context_namespace
```

Comment:
i can't really follow the flow with all 3 of current_namespace, context_namespace, and namespace - are they used correctly everywhere? eg. here https://github.com/cloudify-cosmo/cloudify-common/blob/92926404569ada64b98d5a03cd8ea43ba93ce512/dsl_parser/elements/imports.py#L327